### ISSUE TITLE
YamlElixir do wrong substitution with <<

### ISSUE THREAD

**Issue by deemytch** [KamilLelonek/yaml-elixir#52]
erlang/OTP 24
elixir 1.12
archlinux 

This is yaml file
```
default: &coreAPI
  [publish, disable]: CoreAPI
  [search]: IndexerAPI
storeAPI: &storeAPI
  [fetch, save, update, delete]: StoreAPI

ctrl:
  smartcontract:
    [validate, start, stop, cancel]: BPAPI
    << : *coreAPI
    << : *storeAPI
```

It is processed as expected with ruby 3.0.1
```
[22] pry(main)> cfg = YAML.load_file "config/subst.yml"
=> {"default"=>{["publish", "disable"]=>"CoreAPI", ["search"]=>"IndexerAPI"},
 "storeAPI"=>{["fetch", "save", "update", "delete"]=>"StoreAPI"},
 "ctrl"=>{"smartcontract"=>{["validate", "start", "stop", "cancel"]=>"BPAPI", ["publish", "disable"]=>"CoreAPI", ["search"]=>"IndexerAPI", ["fetch", "save", "update", "delete"]=>"StoreAPI"}}}
```

And not processed with elixir
```
iex(19)> File.read!("config/subst.yml") |> YamlElixir.read_from_string!(atoms: true)
%{
  "ctrl" => %{
    "smartcontract" => %{
      ["validate", "start", "stop", "cancel"] => "BPAPI",
      "<<" => %{["publish", "disable"] => "CoreAPI", ["search"] => "IndexerAPI"}
    }
  },
  "default" => %{
    ["publish", "disable"] => "CoreAPI",
    ["search"] => "IndexerAPI"
  },
  "storeAPI" => %{["fetch", "save", "update", "delete"] => "StoreAPI"}
}
```

Also `atoms: true` won't work.

**Comment by KamilLelonek** [comment]
Can you try that with `{:yaml_elixir, "~> 2.9"}`?

**Comment by KamilLelonek** [comment]
@deemytch Can you try that with `{:yaml_elixir, "~> 2.9"}`?

**Comment by KamilLelonek** [comment]
@deemytch fixed in `{:yaml_elixir, "~> 2.9"}`
